src/database/models.rs

```diff
@@ -224,33 +238,37 @@ impl Subscription {
 pub struct Version {
     pub id: i32,
     pub timestamp: String,
+    pub token: String,
 }
 
 impl Version {
-    pub fn create_new_version(conn: &SqliteConnection) -> Version {
-        insert_into(version_history::table)
-            .default_values()
-            .execute(conn)
-            .expect("New version insertaton failed");
-
-        version_history::table
-            .order(version_history::id.desc())
-            .first::<Version>(conn)
-            .expect("Failed to get new version")
-    }
-
     pub fn current_version(conn: &SqliteConnection) -> Version {
         version_history::table
             .order(version_history::id.desc())
             .first::<Version>(conn)
             .expect("Failed to get latest version")
     }
 
-    pub fn version_from_str(conn: &SqliteConnection, str: &String) -> Version {
+    pub fn get_from_token(conn: &SqliteConnection, token_val: &str) -> Version {
+        let exists = select(exists(
+            version_history::table.filter(version_history::token.eq(token_val)),
+        ))
+        .get_result::<bool>(conn);
```

Comment:
I dont really like the ability to add changes to old versions. Maybe ensure we can only use latest version

Reply:
Well, how would you solve the problem of connecting `POST`, `PUT` and `DELETE` to a single version number?
The idea here is to connect these 3 calls to each other and working under the assumption that the `token` always changes for each run of filter_xml.

Reply:
Personally I probably would not bother making them the same version number, we lose some guarantees due to this change and we don't even need the pretty numbers as its a hidden number. But thats not my point..

What I meant with the comment was that I don't think we should be able to access a version with its token when a newer version exists. So we don't make changes to an old version. So I would like a check that ensures only the latest version can be used.

Reply:
From the last meeting we agreed to just put a check such that we can only get the last version with tokens.